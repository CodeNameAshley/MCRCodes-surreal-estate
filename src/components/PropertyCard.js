import React from "react";
import PropTypes from "prop-types";
import "../styles/PropertyCard.css";

export default function PropertyCard(props) {
  const { title, type, city, bedrooms, bathrooms, price, email } = props;

  return (
    <div className="property-card">
      {title}
      <br />
      {type} - {city}
      <br />
      {Number(bedrooms)}
      <br />
      {Number(bathrooms)}
      <br />
      {Number(price)} bells
      <br />
      {email}
      <br />
    </div>
  );
}

PropertyCard.propTypes = {
  title: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  city: PropTypes.string.isRequired,
  bedrooms: PropTypes.number,
  bathrooms: PropTypes.number,
  price: PropTypes.number,
  email: PropTypes.string.isRequired,
};

PropertyCard.defaultProps = {
  bedrooms: 0,
  bathrooms: 0,
  price: 0,
};
