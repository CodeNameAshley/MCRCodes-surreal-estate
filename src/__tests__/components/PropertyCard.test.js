import React from "react";
import { render, screen } from "@testing-library/react";
import PropertyCard from "../../components/PropertyCard";

// make a test block
// render a component we want to test
// find elements we want to interact with
// interact with those elements
// assert that the results are as expected

test("should have a class name called 'property-card'", () => {
  const { container } = render(<PropertyCard />);

  expect(container.firstChild).toHaveClass("property-card");
});

describe("PropertyCard", () => {
  it("renders correctly", () => {
    const propertyInfo = {
      title: "Rosewood",
      type: "cottage",
      city: "Breezy Hollow",
      bedrooms: "3",
      bathrooms: "2",
      price: "1000",
      email: "ash@acnh.com",
    };
    const { asFragment } = render(
      <PropertyCard
        title={propertyInfo.title}
        type={propertyInfo.type}
        city={propertyInfo.city}
        bedrooms={propertyInfo.bedrooms}
        bathrooms={propertyInfo.bathrooms}
        price={propertyInfo.price}
        email={propertyInfo.email}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
