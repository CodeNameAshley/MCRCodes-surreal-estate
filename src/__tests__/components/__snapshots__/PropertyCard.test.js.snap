// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="property-card"
  >
    Rosewood
    <br />
    cottage - Breezy Hollow
    <br />
    3
    <br />
    2
    <br />
    1000
    <br />
    ash@acnh.com
    <br />
  </div>
</DocumentFragment>
`;
